### ISSUE TITLE
Set up React Testing Library

### ISSUE THREAD

**Issue by raineorshine** [cybersemics/em#930]

**Comment by anmolarora1** [comment]
@raineorshine Here are some updates- 

I tried to test the `window.getSelection` implementation with RTL, and it turns out that it doesn't work with the tests. Though there's no error of any sort because it's mocked, we still don't get the expected values so I couldn't test the selection. Also, the `selectionStart` and `selectionEnd` fields work only for textfields, whereas we're using divs even though [it's type is HTMLInputElement](https://github.com/cybersemics/em/blob/26f84ea5b5a2aaa63e6a1f1197e58e0535b92740/src/components/Editable.tsx#L177), which is a bit misleading. 
Also, the RTL uses JSDOM by default, which mocks the Selection API - https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/selection/Selection-impl.js
That said, the only thing that I need to try out is upgrading the JSDOM as well as the RTL to their latest versions, and see if that changes anything.

**Comment by raineorshine** [comment]
> I tried to test the `window.getSelection` implementation with RTL, and it turns out that it doesn't work with the tests. Though there's no error of any sort because it's mocked, we still don't get the expected values so I couldn't test the selection.

Okay, thanks for finding out.

> Also, the `selectionStart` and `selectionEnd` fields work only for textfields, whereas we're using divs even though [it's type is HTMLInputElement](https://github.com/cybersemics/em/blob/26f84ea5b5a2aaa63e6a1f1197e58e0535b92740/src/components/Editable.tsx#L177), which is a bit misleading.

That's too bad. They should add support for `contenteditable` elements!

> Also, the RTL uses JSDOM by default, which mocks the Selection API - [https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/selection/Selection-impl.js](https://github.com/jsdom/jsdom/blob/master/lib/jsdom/living/selection/Selection-impl.js?rgh-link-date=2020-12-15T18%3A17%3A49Z)

Okay, so they do use JSDOM internally. Good to know.

> That said, the only thing that I need to try out is upgrading the JSDOM as well as the RTL to their latest versions, and see if that changes anything.

It's worth a try. If it doesn't work, I guess we will be unable to test the caret position. At least we can use RTL instead of Enzyme going forward though without losing any functionality.

**Comment by anmolarora1** [comment]
> > That said, the only thing that I need to try out is upgrading the JSDOM as well as the RTL to their latest versions, and see if that changes anything.
> 
> It's worth a try. If it doesn't work, I guess we will be unable to test the caret position. At least we can use RTL instead of Enzyme going forward though without losing any functionality.

@raineorshine I tried updating the libraries but no luck with that. I also tried using [jsdom-sixteen](https://www.npmjs.com/package/jest-environment-jsdom-sixteen) as well but it didn't change anything. 
The implementation for [getSelection](https://github.com/jsdom/jsdom/blob/69f6bf689ba240c29c56f04b6e5d855f86e54a30/lib/jsdom/browser/Window.js#L206) always returns an empty object, irrespective of the scenario that it's used in (or at least our use-cases). So for now, that's the limitation of the library. We can dig more into it and find the cause of that limitation if you prefer that, but I'm not sure if we should spend any further time on this one.

Nevertheless, as you pointed out, we can start using RTL for some of our tests in the future, and it'd be a valuable addition.
